app/src/main/java/co/stonephone/stonecamera/plugins/SettingsTray.kt

```diff
@@ -0,0 +1,171 @@
+                                        style = MaterialTheme.typography.bodyLarge,
+                                        fontWeight = FontWeight.Bold,
+                                        modifier = Modifier.padding(4.dp)
+                                    )
+
+                                    when (setting) {
+                                        is PluginSetting.EnumSetting -> {
+                                            Row(
+                                                modifier = Modifier
+                                                    .fillMaxWidth()
+                                                    .padding(horizontal = 12.dp, vertical = 2.dp),
+                                                horizontalArrangement = Arrangement.SpaceBetween
+                                            ) {
+                                                setting.options.map { option ->
+                                                    Box(modifier = Modifier
+                                                        .then(modifier)
+                                                        .clickable {
+                                                            viewModel.setSetting(
+                                                                setting.key, option
+                                                            )
+                                                        }) {
+                                                        setting.render(option, option == value)
+                                                    }
+                                                }
+                                            }
+                                        }
+
+                                        is PluginSetting.ScalarSetting -> {
+                                            Slider(
+                                                value = value as Float,
+                                                onValueChange = { newValue ->
+                                                    viewModel.setSetting(setting.key, newValue)
+                                                },
+                                                valueRange = setting.minValue..setting.maxValue,
+                                                modifier = modifier
+                                            )
+                                        }
+
+                                        is PluginSetting.CustomSetting -> setting.customRender(
+                                            viewModel, value
+                                        ) { value ->
+                                            viewModel.setSetting(setting.key, value)
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
```

Comment:
Somewhat tongue-in-cheek, 9 levels of nesting seems excessive. Of course, the code needs each of these closing braces, I'm more curious why we end up so deeply nested to provide the functionality.